neo/Trie/MPT/MPTTrie.Put.cs

```diff
@@ -0,0 +1,173 @@
+            var path = ToNibbles(key);
+            if (value.Length == 0)
+            {
+                return TryDelete(ref root, path);
+            }
+            var n = new LeafNode(value);
+            return Put(ref root, path, n);
+        }
+
+        private bool Put(ref MPTNode node, ReadOnlySpan<byte> path, MPTNode val)
+        {
+            switch (node)
+            {
+                case LeafNode leafNode:
+                    {
+                        if (val is LeafNode v)
+                        {
+                            if (path.IsEmpty)
+                            {
+                                if (leafNode.Value.AsSpan().SequenceEqual(v.Value.AsSpan()))
+                                    return true;
+                                node = v;
+                                db.Put(v);
+                                return true;
+                            }
+                            var branch = new BranchNode();
+                            branch.Children[BranchNode.ChildCount - 1] = leafNode;
+                            Put(ref branch.Children[path[0]], path.Slice(1), v);
+                            db.Put(branch);
+                            node = branch;
+                            return true;
+                        }
+                        return false;
+                    }
+                case ExtensionNode extensionNode:
+                    {
+                        if (path.StartsWith(extensionNode.Key))
+                        {
+                            if (Put(ref extensionNode.Next, path.Slice(extensionNode.Key.Length), val))
+                            {
+                                extensionNode.SetDirty();
+                                db.Put(extensionNode);
+                                return true;
+                            }
+                            return false;
+                        }
+                        var prefix = CommonPrefix(extensionNode.Key, path);
+                        var pathRemain = path.Slice(prefix.Length);
+                        var keyRemain = extensionNode.Key.AsSpan().Slice(prefix.Length);
+                        var child = new BranchNode();
+                        MPTNode grandChild1 = HashNode.EmptyNode;
+                        MPTNode grandChild2 = HashNode.EmptyNode;
+                        if (keyRemain.Length == 1)
+                        {
+                            child.Children[keyRemain[0]] = extensionNode.Next;
+                        }
+                        else
+                        {
+                            var exNode = new ExtensionNode
+                            {
+                                Key = keyRemain.Slice(1).ToArray(),
+                                Next = extensionNode.Next,
+                            };
+                            db.Put(exNode);
+                            child.Children[keyRemain[0]] = exNode;
```

Comment:
if `keyRemain` it's 0 length?

Reply:
`keyRemain` won't be 0. If it is, we will already handle here:
https://github.com/KickSeason/neo/blob/355d7bcdaac5e97af19c61bc98f98df6a425ab93/neo/Trie/MPT/MPTTrie.Put.cs#L66

Reply:
If `extensionNode.Key` it's longer than 0, and path it's 0?

Reply:
If so, `pathRemain.Length` is 0, we have check here: https://github.com/KickSeason/neo/blob/5f49bd8e803f2d24817420495df903d637738f43/neo/Trie/MPT/MPTTrie.Put.cs#L93

And `prefix.Length` is 0, we check https://github.com/KickSeason/neo/blob/5f49bd8e803f2d24817420495df903d637738f43/neo/Trie/MPT/MPTTrie.Put.cs#L104